test/integration/cls.test.js

```diff
@@ -164,5 +164,17 @@ if (current.dialect.supports.transactions) {
     it('CLS namespace is stored in Sequelize._cls', function() {
       expect(Sequelize._cls).to.equal(this.ns);
     });
+
+    it('promises returned by sequelize.query are correctly patched', function() {
```

Comment:
Needs a real world test, create row under transaction, try to access that row in nested context assuming cls context is passed

Reply:
Or may be take inspiration from #9383 (2nd test looks neat)

Reply:
Actually it's already almost exactly the 2nd test from #9383.
I just removed the first two `.then()`, which are useless because the first `sequelize.query()` already returns a non-patched promise.